File: src/ProjectR.Scripting/content/scripts/spells/lua/HiddenPower.lua
function GetTargetType()
    return Myself
end

function GetName()
  return "Hidden Power"
end

function GetDescription()
  return "Heighten your combat ability for speed.\nIncreases AD & DEF, decreases SPD."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .8
end

function GetDelay()
  return .8
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  d:BuffStat(AD, .75)
  d:BuffStat(DEF, .75)
  d:BuffStat(SPD, -.75)
end

File: src/ProjectR.Scripting/content/scripts/spells/HiddenPower.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class HiddenPower : SpellScriptBase
    {
        public override string Name { get { return "Hidden Power"; } }
        public override string Description { get { return "Heighten your combat ability for speed.\nIncreases AD & DEF, decreases SPD."; } }

        public override TargetType TargetType { get { return TargetType.Myself; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return true; } }

        public override double MPCost { get { return 80; } }
        public override double Delay { get { return .8; } }

        public override void SpellEffect(ICharacter caster)
        {
            BuffStat(Stat.AD, .75);
            BuffStat(Stat.DEF, .75);
            BuffStat(Stat.SPD, -.75);
        }
    }
}